test/Parse/raw_string.swift

```diff
@@ -68,6 +68,37 @@ _ = ##"""
       """##
 // CHECK: "a raw string with \"\"\" in it"
 
+_ = #"""#
+// CHECK: "\""
```

Comment:
Here's a tricky situation that I want to make sure works in the future. Hopefully, this makes sense to you:

For now, `diagnoseZeroWidthMatchAndAdvance` doesn't actually diagnose zero-width spaces. It's meant to be clever so that eventually `"[U+200B]"[U+200B]"` (where `[U+200B]` is the zero-width space) is lexed as `"""` for the purposes of multiline string literals.

For your purposes here, _even if that eventually becomes implemented_, it's important that `#"[U+200B]"[U+200B]"#` is still lexed exactly as it is right now. That is, the string should be equal to `"\u{200B}\"\u{200B}"` and not just `"\""`. If we test for that now, it's trivial, but if someone implements `diagnoseZeroWidthMatchAndAdvance` in the future and it breaks this, then they will be alerted to that fact. We can even leave future people a note explaining this.

Reply:
That's a great point! I read about diagnoseZeroWidthMatchAndAdvance in SR-8678, so I understand your point. 

Adding the test case and a short note now.